File: src/main/scala/com/wixpress/embed/mysql/config/MysqldConfig.scala
package com.wixpress.embed.mysql.config

import de.flapdoodle.embed.process.config.{ExecutableProcessConfig, ISupportConfig}
import de.flapdoodle.embed.process.distribution.IVersion

/**
 * @author viliusl
 * @since 18/09/14
 */
class MysqldConfig(
  version: IVersion,
  val port: Int = 3306) extends ExecutableProcessConfig(version, new MysqldSupportConfig()) {
}

class MysqldSupportConfig extends ISupportConfig {
  override def messageOnException(context: Class[_], exception: Exception): String = ""
  override def getName: String = "mysqld"
  override def getSupportUrl: String = "https://github.com/wix/wix-embedded-mysql/issues"
}

File: src/main/java/com/wixpress/embed/mysql/config/MysqldConfig.java
package com.wixpress.embed.mysql.config;

import de.flapdoodle.embed.process.config.ExecutableProcessConfig;
import de.flapdoodle.embed.process.config.ISupportConfig;
import de.flapdoodle.embed.process.distribution.IVersion;

/**
 * @author viliusl
 * @since 27/09/14
 */
public class MysqldConfig extends ExecutableProcessConfig {

    private int port = 3306;

    public MysqldConfig(IVersion version) {
        super(version, new MysqldSupportConfig());
    }

    public MysqldConfig(IVersion version, int port) {
        super(version, new MysqldSupportConfig());
        this.port = port;
    }

    public int getPort() { return port; }
    public int getTimeout() { return 6000; }


    public static class MysqldSupportConfig implements ISupportConfig {
        @Override
        public String getName() {
            return "mysqld";
        }

        @Override
        public String getSupportUrl() {
            return "https://github.com/wix/wix-embedded-mysql/issues";
        }

        @Override
        public String messageOnException(Class<?> context, Exception exception) {
            return null;
        }
    }
}